qiskit/circuit/gate.py

```diff
@@ -58,9 +58,10 @@ def power(self, exponent):
         Raises:
             CircuitError: If Gate is not unitary
         """
+        from qiskit.quantum_info.operators import Operator
         from qiskit.extensions.unitary import UnitaryGate  # pylint: disable=cyclic-import
         # Should be diagonalized because it's a unitary.
-        decomposition, unitary = schur(self.to_matrix(), output='complex')
+        decomposition, unitary = schur(Operator(self).data, output='complex')
```

Comment:
Could we just make this the default implementation of `to_matrix` for the Gate class?

Reply:
potentially - but the reason it has not been done yet is because of global phase ambiguities in the definitions.